setup/steps/09_deploy_via_modelservice.py

```diff
@@ -0,0 +1,693 @@
+    
+    prefill_extra_pod_config = ev.get("vllm_modelservice_prefill_extra_pod_config", "")
+    prefill_extra_container_config = ev.get("vllm_modelservice_prefill_extra_container_config", "")
+    prefill_extra_volume_mounts = ev.get("vllm_modelservice_prefill_extra_volume_mounts", "")
+    prefill_extra_volumes = ev.get("vllm_modelservice_prefill_extra_volumes", "")
+    
+    # Environment variables to YAML
+    envvars_to_yaml = ev.get("vllm_common_envvars_to_yaml", "")
+    
+    # Read the rules file content
+    rules_content = ""
+    if rules_file.exists():
+        rules_content = rules_file.read_text().rstrip()
+    
+    # Build the complete YAML structure exactly matching the bash script
+    yaml_content = f"""fullnameOverride: {fullname_override}
+multinode: {multinode}
+
+modelArtifacts:
+  uri: {model_uri}
+  size: {model_size}
+  authSecretName: "llm-d-hf-token"
+  name: {model_name}
+
+routing:
+  servicePort: {service_port}
+  parentRefs:
+    - group: gateway.networking.k8s.io
+      kind: Gateway
+      name: infra-{release}-inference-gateway
+  proxy:
+    image: "{proxy_image}"
+    secure: false
+    connector: {proxy_connector}
+    debugLevel: {proxy_debug_level}
+  inferenceModel:
+    create: {inference_model_create}
+  inferencePool:
+    create: {inference_pool_create}
+    name: {model_id_label}-gaie
+  httpRoute:
+    create: {route_enabled}
+    rules:
+    - backendRefs:
+      - group: inference.networking.x-k8s.io
+        kind: InferencePool
+        name: {model_id_label}-gaie
+        port: 8000
+        weight: 1
+      matches:
+      - path:
+          type: PathPrefix
+          value: /{model_id}/
+      filters:
+      - type: URLRewrite
+        urlRewrite:
+          path:
+            type: ReplacePrefixMatch
+            replacePrefixMatch: /
+    {rules_content}
+
+  epp:
+    create: {epp_create}
+
+decode:
+  create: {decode_create}
+  replicas: {decode_replicas}
+  acceleratorTypes:
+      labelKey: {affinity_key}
+      labelValues:
+        - {affinity_value}
+  parallelism:
+    data: {decode_data_parallelism}
+    tensor: {decode_tensor_parallelism}
+  annotations:
+      {functions_add_annotations("LLMDBENCH_VLLM_COMMON_ANNOTATIONS")}
+  podAnnotations:
+      {add_pod_annotations("LLMDBENCH_VLLM_MODELSERVICE_DECODE_PODANNOTATIONS")}
+  {functions_add_config(decode_extra_pod_config, 2, "extraConfig")}
+  containers:
+  - name: "vllm"
+    mountModelVolume: {str(mount_model_volume).lower()}
+    image: "{main_image}"
+    modelCommand: {decode_model_command}
+    {add_command(decode_model_command)}
+    args:
+      {add_command_line_options(decode_extra_args)}
+    env:
+      - name: VLLM_NIXL_SIDE_CHANNEL_HOST
+        valueFrom:
+          fieldRef:
+            fieldPath: status.podIP
+      {functions_add_additional_env_to_yaml(envvars_to_yaml)}
+    ports:
+      - containerPort: {decode_inference_port}
+      - containerPort: 5557
```

Comment:
This is insufficient. The metrics collection requires a `name`. There is an env variable that should be looked at. An example is in `scenarios/examples/inference-scheduling.sh`.  I think it should be possible to reuse `add_config()`.

```
export LLMDBENCH_VLLM_MODELSERVICE_EXTRA_CONTAINER_CONFIG=$(mktemp)
cat << EOF > ${LLMDBENCH_VLLM_MODELSERVICE_EXTRA_CONTAINER_CONFIG}
ports:
  - containerPort: 5557
    protocol: TCP
  - containerPort: 8200
    name: metrics
    protocol: TCP
EOF
```
Note that in this example, the `8200` should probably be replaced by `REPLACE_ENV_VLLM_MODELSERVICE_DECODE_INFERENCE_PORT`